awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -588,3 +589,34 @@ def is_job_done(self, job_status):
             return False
         else:
             return True
+
+    def wait_on_url(self, url, object_name, object_type, timeout=30, interval=10):
+        # Grab our start time to compare against for the timeout
+        start = time.time()
+        result = self.get_endpoint(url)
+        while not result['json']['finished']:
+            # If we are past our time out fail with a message
+            if timeout and timeout < time.time() - start:
+                self.json_output['msg'] = 'Monitoring of {0} "{1}" aborted due to timeout'.format(object_type, object_name)
+                self.wait_output(result)
+                self.fail_json(**self.json_output)
+
+            # Put the process to sleep for our interval
+            time.sleep(interval)
+
+            result = self.get_endpoint(url)
+            self.json_output['status'] = result['json']['status']
+
+        # If the job has failed, we want to raise a task failure for that so we get a non-zero response.
+        if result['json']['failed']:
+            self.json_output['msg'] = 'The {0} "{1}" failed'.format(object_type, object_name)
```

Comment:
The one `awx-ansible-module` CI test that is erroring out I think directly comes back to this part of the code.  

The error message is: `AssertionError: assert {'msg': 'The job "1" failed'} != {'msg': 'Job with id 1 failed'}`, which makes me think that we either need to change the expected output of the `test_job_wait_failed ` unit test in `/awx_devel/awx_collection/test/awx/test_job.py`, or the format of the error message in this part of the module utils.

Reply:
I could change it to the old format, or update the Test. I like the idea of the new format, and either change is easy, I think its down to, what do we prefer? Keeping with legacy format, the one, or something else?

Reply:
@beeankha updated to pass tests and look pretty for everything but tower_job_wait.

Reply:
Updated to work for legacy and have nice output for other.